dev/com.ibm.ws.jca_fat_example_anno/test-resourceadapter/src/com/ibm/example/jca/anno/IndexedRecordImpl.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.example.jca.anno;
+
+import java.util.ArrayList;
+
+import jakarta.resource.cci.IndexedRecord;
+
+/**
+ * Example IndexedRecord where entries are kept in a ArrayList.
+ */
+public class IndexedRecordImpl<K> extends ArrayList<K> implements IndexedRecord<K> {
```

Comment:
Could we also test a declared implementation?

```suggestion
public class IndexedRecordImpl<String> extends ArrayList<String> implements IndexedRecord<String> {
```

Reply:
It is possible, I can add another test if you feel strongly, but personally don't think it buys us much.